src/main/java/com/google/devtools/build/lib/authandtls/AuthAndTLSOptions.java

```diff
@@ -61,17 +61,32 @@
   )
   public String googleCredentials;
 
+  @Deprecated
   @Option(
     name = "tls_enabled",
     defaultValue = "false",
     documentationCategory = OptionDocumentationCategory.UNCATEGORIZED,
+    deprecationWarning = "Deprecated. Please specify a valid protocol in the URL (https or grpcs).",
     effectTags = {OptionEffectTag.UNKNOWN},
     help =
-        "Specifies whether to use TLS for remote execution/caching and the build event service"
-            + " (BES)."
+        "DEPRECATED. Specifies whether to use TLS for remote execution/caching and "
+            + "the build event service (BES)."
   )
   public boolean tlsEnabled;
 
+  @Option(
+    name = "incompatible_tls_enabled_removed",
+    defaultValue = "false",
+    documentationCategory = OptionDocumentationCategory.UNCATEGORIZED,
+    effectTags = {OptionEffectTag.UNKNOWN},
+    metadataTags = {
+        OptionMetadataTag.INCOMPATIBLE_CHANGE,
+        OptionMetadataTag.TRIGGERED_BY_ALL_INCOMPATIBLE_CHANGES
+    },
+    help = "If set to true, bazel will handle --tls_enabled as a not existing flag."
```

Comment:
Please create an github issue for this incompatible flag and reference it in the help text. The github issue should state in which release this flag will be flipped and how to migrate.

Reply:
done